sdk/identity/Azure.Identity/src/ManagedIdentityClient.cs

```diff
@@ -227,14 +227,20 @@ private bool ImdsAvailable(CancellationToken cancellationToken)
 
                     return true;
                 }
-                // we only want to handle the case when the imdsTimeout resulted in the request being cancelled.
+                // we only want to handle the case when the imdsTimeout resulted in the request being canceled
                 // this indicates that the request timed out and that imds is not available.  If the operation
-                // was user cancelled we don't wan't to handle the exception so s_identityAvailable will
+                // was user canceled we don't want to handle the exception so s_identityAvailable will
                 // remain unset, as we still haven't determined if the imds endpoint is available.
                 catch (OperationCanceledException) when (!cancellationToken.IsCancellationRequested)
                 {
                     return false;
                 }
+                // in the case that the request failed in a manner that we didn't receive any response, take this
+                // to indicate that the imds endpoint is not available.
+                catch (Exception)
```

Comment:
Doesn't this catch handler run when `OperationCanceledException` is thrown but `cancellationToken.IsCancellationRequested` is true?  If so - I don't think that's expected (it would mean a well timed canceled operation would poison things)?

For example, this program:

```
        static void Main(string[] args)
        {
            CancellationTokenSource cs = new CancellationTokenSource();
            cs.Cancel();
            try
            {               
                throw new OperationCanceledException();
            } catch (OperationCanceledException) when (!cs.Token.IsCancellationRequested)
            {
                Console.WriteLine("OperationCanceledException handler");
            } catch (Exception)
            {
                Console.WriteLine("Exception handler");
            }

            Console.WriteLine("Done");
        }
```

Prints

```
Exception handler
Done
```